storm/check.py

```diff
@@ -0,0 +1,573 @@
+        :param interval: Interval in seconds for reported metrics
+        :type interval: int
+        """
+        if len(topology_stats) > 0:
+            name = topology_name.replace('.', '_').replace(':', '_')
+            tags = ['topology:{}'.format(name)]
+            r = {'bolts': [], 'spouts': []}
+            for k in r.keys():
+                for s in _g(topology_stats, [], None, k):
+                    k_name = _g(s, 'unknown', None, 'id').replace('.', '_').replace(':', '_')
+                    k_tags = tags + ['{}:{}'.format(k, k_name)]
+                    for sc in ['acked', 'complete_ms_avg', 'emitted', 'executed', 'executed_ms_avg', 'failed',
+                               'process_ms_avg', 'transferred']:
+                        for ks in _g(s, [], None, sc):
+                            stream_id = _g(ks, 'unknown', None, 'stream_id')
+                            ks_tags = k_tags + ['stream:{}'.format(stream_id)]
+                            component_id = _g(ks, None, None, 'component_id')
+                            if component_id:
+                                ks_tags.append('component:{}'.format(component_id))
+
+                            component_value = _g(ks, 0.0, _float, 'value')
+                            if component_value is not None:
+                                # will make stats like these two examples
+                                # storm.topologyStats.metrics.spouts.last_60.emitted
+                                # storm.topologyStatus.metrics.bolts.last_60.acked
+                                self.report_histogram(
+                                    'storm.topologyStats.metrics.{}.last_{}.{}'.format(k, interval, sc),
+                                    component_value,
+                                    tags=ks_tags, additional_tags=self.additional_tags
+                                )
+
+    def report_gauge(self, metric, value, tags, additional_tags=list()):
+        """ Report the Gauge Metric.
+
+        :param metric:
+        :param value:
+        :param tags:
+        :param additional_tags:
+        :return:
+        """
+        all_tags = tags + [
+            'env:{}'.format(self.environment_name),
+            'environment:{}'.format(self.environment_name)] + additional_tags
+        self.gauge(
+            metric=metric,
+            value=value,
+            tags=all_tags
+        )
+
+    def report_histogram(self, metric, value, tags, additional_tags=list()):
+        """ Report the Histogram Metric.
+
+        :param metric:
+        :param value:
+        :param tags:
+        :param additional_tags:
+        :return:
+        """
+        all_tags = tags + [
+            'env:{}'.format(self.environment_name),
+            'environment:{}'.format(self.environment_name)] + additional_tags
+        self.histogram(
+            metric=metric,
+            value=value,
+            tags=all_tags
+        )
+
+    def update_from_config(self, instance):
```

Comment:
The default values for `nimbus_server`, `environment_name` and `intervals` should be class constants.